### PR TITLE
[DQM] Prompt Matrix reconfiguration

### PR DESCRIPTION
**Pull request by jfernan2**
Changes for Prompt Matrix DQM reduction:

Reduced:
- L1TMon to SingleMuon, SingleElectron/EG and ZB
- TAU to SingleMuon, SingleElectron/EG and TAU
- TRK to SingleMuon, SingleElectron/EG, DoubleMuon, JetHT, JetMET and ZB, by creating commonReduced which excludes TRK
- CTPPS to DoubleMuon, SingleElectron/EG y ZB

# Replay Request

**Requestor**  
PPD-DQM

**Describe the configuration**  
* Release: CMSSW_12_1_X and PR https://github.com/cms-sw/cmssw/pull/35605
* Run: 317696
* GTs:
        expressGlobalTag: 120X_dataRun3_Express_Candidate_2021_09_30_18_52_55
        promptrecoGlobalTag: 120X_dataRun3_Prompt_Candidate_2021_09_30_19_06_33
        alcap0GlobalTag: 120X_dataRun3_Prompt_Candidate_2021_09_30_19_06_33

* Additional changes:
This PR goes in sync with: cms-sw/cmssw#35605
   
**Purpose of the test**  

Not sure if a replay is needed. PPD started a campaign to reduce the DQM load on Prompt matrix. This PR reduces the number of sequences as agreed on:

https://docs.google.com/presentation/d/1AF65xzq7T70Yt-0o_OV47YTFBv-kbvHZDlRmUSCZsAw/edit#slide=id.geedc047305_0_0

**T0 Operations HyperNews thread**   
https://hypernews.cern.ch/HyperNews/CMS/get/tier0-Ops/2314.html

Thanks

### CODE REVIEW

**Comment by germanfgv** [comment]
@jfernan2 I'm a bit confused. If we need cms-sw/cmssw#35605 to properly test this configuration, then that PR should be already merge into a release. We can only test CMSSW code that's available in /cvmfs/. 

If that's not the case, then what CMSSW release would you like us to use?

**Comment by jfernan2** [comment]
@germanfgv  PR #35605 was merged in CMSSW_12_1_0_pre5
https://github.com/cms-sw/cmssw/releases/tag/CMSSW_12_1_0_pre5
So, if you could perform the replay at some point it is apreciated
Thanks

**Comment by tvami** [comment]
Hi @jfernan2 I think we wanted to wait for `CMSSW_12_1_0` to come out, i.e. tomorrow

**Comment by qliphy** [comment]
@tvami @jfernan2 @germanfgv   CMSSW_12_1_0 is now ready.

**Comment by germanfgv** [comment]
@jfernan2 @tvami I updated the CMSSW version. Can you check GTs and runs before trigerring the replay?

**Comment by francescobrivio** [comment]
@germanfgv I think the best way to the cpu usge reduction with the new DQM Matrix is to use a 2018 pp run, as specified in: https://github.com/dmwm/T0/pull/4619#issuecomment-939849057
So you should update in the configuration:
 - run number
 - GTs
 - different scenarios

**Comment by germanfgv** [comment]
@francescobrivio I made the necessary changes. I'll start the replay.

**Comment by francescobrivio** [comment]
looks good! thanks @germanfgv !

**Comment by germanfgv** [comment]
run replay please

**Comment by germanfgv** [comment]
@francescobrivio it looks like there is an issue with the GTs. We are getting the following error, for both Express and Prompt:
```
An exception of category 'NoRecord' occurred while
   [0] Processing global begin Run run: 317696
   [1] Prefetching for module TotemTimingDQMSource/'totemTimingDQMSource'
   [2] Prefetching for EventSetup module CTPPSGeometryESModule/'ctppsGeometryESModule'
   [3] Calling method for EventSetup module CTPPSGeometryESModule/'ctppsGeometryESModule'
   [4] While getting dependent Record from Record VeryForwardRealGeometryRecord
Exception Message:
No "VeryForwardIdealGeometryRecord" record found in the EventSetup.

 The Record is delivered by an ESSource or ESProducer but there is no valid IOV for the synchronization value.
 Please check 
   a) if the synchronization value is reasonable and report to the hypernews if it is not.
   b) else check that all ESSources have been properly configured.
```
Should we be using different GTs?

**Comment by tvami** [comment]
Hi @germanfgv 
yes, this `VeryForwardIdealGeometryRecord` was added recently.
Let's use these GTs
 * Express: 121X_dataRun3_Express_v11
 * Prompt: 121X_dataRun3_Prompt_v10

For 2018
 * Express: 121X_dataRun3_Express_Candidate_2021_11_06_15_02_30
 * Prompt: 121X_dataRun3_Prompt_Candidate_2021_11_06_15_02_45

**Comment by germanfgv** [comment]
run replay please

**Comment by germanfgv** [comment]
Now  we have another ESSource error.
```
CMSSW Return code: 8001

CRITICAL:root:Error message: An exception of category 'EventSetupConflict' occurred while
   [0] Calling beginJob
Exception Message:
two EventSetup Sources want to deliver type="PPSAssociationCuts" label=""
 from record PPSAssociationCutsRcd. The two providers are 
1) type="PPSAssociationCutsESSource" label="ppsAssociationCutsESSource"
2) type="PoolDBESSource" label="GlobalTag"
Please either
   remove one of these Sources
   or find a way of configuring one of them so it does not deliver this data
   or use an es_prefer statement in the configuration to choose one.
```

**Comment by tvami** [comment]
Yeah the issue is that https://github.com/cms-sw/cmssw/pull/35766 only got merged in 12_2_X

So let's revert to a prev GT
  * Express: 121X_dataRun3_Express_v10
  * Prompt: 121X_dataRun3_Prompt_v9

For 2018:
  * Express: 121X_dataRun3_Express_Candidate_2021_11_07_15_29_55
  * Prompt: 121X_dataRun3_Prompt_Candidate_2021_11_07_15_30_12

**Comment by germanfgv** [comment]
run replay please

**Comment by germanfgv** [comment]
The current replays has got paused  Reco jobs with the following error
```
An exception of category 'HCALDQM' occurred while
   [0] Processing  Event run: 317696 lumi: 4 event: 18806 stream: 2
   [1] Running path 'dqmoffline_1_step'
   [2] Calling method for module TPTask/'tpTask'
Exception Message:
TPTask::Collection HcalTrigPrimDigiCollection isn't available: valHcalTriggerPrimitiveDigi
```
I'll provide logs and tarballs in a few minutes

**Comment by tvami** [comment]
> The current replays has got paused Reco jobs with the following error
> 
> ```
> An exception of category 'HCALDQM' occurred while
>    [0] Processing  Event run: 317696 lumi: 4 event: 18806 stream: 2
>    [1] Running path 'dqmoffline_1_step'
>    [2] Calling method for module TPTask/'tpTask'
> Exception Message:
> TPTask::Collection HcalTrigPrimDigiCollection isn't available: valHcalTriggerPrimitiveDigi
> ```
> 
> I'll provide logs and tarballs in a few minutes

This is on the DQM friends this time, @jfernan2

**Comment by germanfgv** [comment]
In case you havent checked the HN thread, you can reproduce the error running the PSet that can be found in:
```
/afs/cern.ch/user/c/cmst0/public/PausedJobs/DQMseq/job_9868
```

**Comment by jfernan2** [comment]
Thanks @germanfgv  but I am afraid I am not able to reproduce the error since the PSet.py you pointed gives another error[1]

In any case, since this PR is not adding additional HCAL DQM sequences, but in any case removing them, may I ask what exact cmsDriver or configuration is the replay running? 

The stream that crashes (according to the log an PSet you sent) has as PD JetHT, which I am not touching in this PR for the replay config:

https://github.com/dmwm/T0/blob/47ad2b08266a27189d608158e56110d2caad8533/etc/ReplayOfflineConfiguration.py#L622-L633

So, either the replay config was already wrong or I do not understand what we are exactly running. Thanks

[1] 08-Nov-2021 17:19:22 CET  Successfully opened file root://eoscms.cern.ch//eos/cms/tier0/store/backfill/1/data/Tier0_REPLAY_2021/JetHT/RAW/v2111071948/000/317/696/00000/DC4F0B2F-0240-EC11-8101-A4BF0112DEEC.root?eos.app=cmst0
2021-11-08 17:19:50.008554: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  SSE4.1 SSE4.2 AVX AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
----- Begin Fatal Exception 08-Nov-2021 17:20:25 CET-----------------------
An exception of category 'FileOpenError' occurred while
   [0] Calling StorageFactory::open()
   [1] Calling File::sysopen()
Exception Message:
Failed to open the file 'SKIMStreamJetHTJetPlusHOFilter.root'
   Additional Info:
      [a] open() failed with system error 'Permission denied' (error code 13)
----- End Fatal Exception -------------------------------------------------
%MSG-w StatisticsSenderService:  AfterModEndJob 08-Nov-2021 17:20:25 CET EndJob
fully closed: /store/backfill/1/data/Tier0_REPLAY_2021/JetHT/RAW/v2111071948/000/317/696/00000/DC4F0B2F-0240-EC11-8101-A4BF0112DEEC.root

**Comment by germanfgv** [comment]
@jfernan2 Sorry for the delay replying to this.  We don't use cmsDriver, so I cannot give you the command. I can generate the `process_funcArgs.json`. Files from this replay have been cleared up, but I'll run another one with the same config.

In any case, I'm not sure it fails opening the file. It is there in EOS:
```
$ ll /eos/cms/tier0/store/backfill/1/data/Tier0_REPLAY_2021/JetHT/RAW/v2111071948/000/317/696/00000/DC4F0B2F-0240-EC11-8101-A4BF0112DEEC.root
-rw-r--r--. 1 cmst0 zh 3032440649 Nov  7 20:50 /eos/cms/tier0/store/backfill/1/data/Tier0_REPLAY_2021/JetHT/RAW/v2111071948/000/317/696/00000/DC4F0B2F-0240-EC11-8101-A4BF0112DEEC.root
```

**Comment by jfernan2** [comment]
Thanks a lot @germanfgv 
I see you added /afs/cern.ch/user/c/cmst0/public/PausedJobs/DQMseq/job_9868/generatedConfig/   folder but I do not see any process_funcArgs.json file there or in the bz2 archive

Anyway, I manage to overcome my file reading problem and I am able to reproduce the error, however I am puzzled, since this error should be there already present in previous replays because, as I said, ProdOfflineConfiguration.py has this sequence too

https://github.com/dmwm/T0/blob/47ad2b08266a27189d608158e56110d2caad8533/etc/ProdOfflineConfiguration.py#L651

Indeed, other PDs should be crashing too. Is only JetHT crashing? "@hcal" sequence is present in many other PDs. Is the replay running other PDs apart from JetHT? From the bot's settings it looks to me that other PDs should be crashing too:

https://cmst0.web.cern.ch/CMST0/tier0/offline_config/ReplayOfflineConfiguration_047.php

On the other hand, it is difficult for me to fix the issue without knowing how the Tier0 is extracting the PSet config

**Comment by germanfgv** [comment]
No only JetHT. Other PD failing are  Commissioning, HLTPhysics, HcalNZS and MET. You can find another example PSet, along with  `process_funcArgs.json`

```
/afs/cern.ch/user/c/cmst0/public/PausedJobs/DQMseq/job_9484/generatedConfig/job/WMTaskSpace/cmsRun1
```

**Comment by jfernan2** [comment]
Thanks @germanfgv 
That makes sense. Indeed the offending module has been there since several years, so the crash is not related to the change tested in this PR.

May it be that we have not made any replay with ppScenario : ppEra_Run2_2018 since the end of the Run2? I am contacting the Hcal and L1T DQM developers who may shred some light into this, I do not understand why the missing collection they are trying to monitor in DQM (valHcalTriggerPrimitiveDigis) is not present any longer in RECO....

**Comment by francescobrivio** [comment]
> May it be that we have not made any replay with ppScenario : ppEra_Run2_2018 since the end of the Run2?

@jfernan2 We actually did several replays with `ppScenario = "ppEra_Run2_2018"` (see for example https://github.com/dmwm/T0/pull/4622). What is different here?

**Comment by jfernan2** [comment]
Thanks @francescobrivio 
Indeed in 12_0_X where previous replay was done, the config is working. So, I have found out the reason of the problem which is also present in master.

I understand I need to make a new PR for this and wait for another release to be built with it for the replay to take place at Tier0, right @germanfgv ? However, since 12_1_X cycle is death, should we move to 12_2_X instead? @qliphy @perrotta  could you advise on this?

Thank you very much

**Comment by tvami** [comment]
Hi @jfernan2 yes please make a PR to master. And yes T0 needs a new release or pre-release. We'll actually build the 12_2_0_pre2 soon, so if your PR goes in fast we could use that pre-release for this test.

**Comment by qliphy** [comment]
@jfernan2 @tvami   Yes, 12_2_0_pre2 will be built this week as soon as the IB test with mkFit updates will be done.

**Comment by jfernan2** [comment]
Thanks @tvami @qliphy 
I found a workaround which avoids a new PR in CMSSW with low computing cost: since changes in hcal sequence now require L1TMon to be run in front in order to produce the missing collection valHcalTriggerPrimitiveDigis which is monitored by HCAL DQM, we must force to run them hand by hand.

@germanfgv could you please trigger again the replay with the last commit I made to this PR? I believe now all is fine. I am sorry for the inconvenience

**Comment by tvami** [comment]
run replay please

**Comment by jfernan2** [comment]
run replay please

**Comment by tvami** [comment]
run replay please

**Comment by tvami** [comment]
This test has failed.

> /tmp/jenkins8292252753278809325.sh: line 38: syntax error near unexpected token `else'

I'm not sure what we can do w/o @germanfgv about this

**Comment by germanfgv** [comment]
The replay tryied to start but there is something worng with the config file after today's commits. I'm checking what it is

File: etc/ReplayOfflineConfiguration.py [hunk 1]
```diff
@@ -889,7 +895,22 @@
                write_dqm=True,
                write_miniaod=True,
                write_aod=True,
-               dqm_sequences=["@common", "@ecal", "@jetmet", "@hcal", "@L1TEgamma"],
+               dqm_sequences=["@common", "@ecal", "@jetmet", "@L1TMon", "@hcal", "@L1TEgamma"],
+               alca_producers=["TkAlMinBias"],
+               physics_skims=["LogError", "LogErrorMonitor"],
+               scenario=ppScenario)
+
+DATASETS = ["HLTPhysics0", "HLTPhysics1", "HLTPhysics2",
```
Comment (germanfgv) [inline etc/ReplayOfflineConfiguration.py:903]: Not sure where this comes from, but HLTPhysics datasets are duplicated. That's why the replay was not starting, I manually fixed it in the container. The replay is now running, although Jenkins thinks it already failed.

**Comment by jfernan2** [comment]
Thanks @germanfgv 
I probably screw it up when trying to fix conflicts right after my last commit since the version from my branch was not in sync with master.

Anyway, the right line is the second:
dqm_sequences=["@common", "@ecal", "@jetmet", "@L1TMon", "@hcal", "@L1TEgamma"],

I am sorry for the inconvenience. I will fix it with another commit once the replay is over to not confuse it even more

**Comment by germanfgv** [comment]
One MET Reco jobs is using too much memory:
```
PerformanceKill (Exit Code: 50660)
Error in CMSSW step cmsRun1
Number of Cores: 8
Job has exceeded maxPSS: 12300 MB
Job has PSS: 13540 MB
```

I put logs and tarball in `/afs/cern.ch/user/c/cmst0/public/PausedJobs/DQMseq/job_9738/tarball`

**Comment by germanfgv** [comment]
I'll increase the memory for the job and retry to continue with the test, but please take a look at the logs.

**Comment by germanfgv** [comment]
Other that the issues with  memory, the replay finished withput problems

**Comment by jfernan2** [comment]
Thanks @germanfgv 
I was not able to reproduce the momery porblems running on lxplus, any advice?

**Comment by germanfgv** [comment]
> Thanks @germanfgv I was not able to reproduce the memory problems running on lxplus, any advice?

@jfernan2  How did you monitor your memory usage? what was the maximum amount you reached?

**Comment by jfernan2** [comment]
@germanfgv  I got the usual output from cmsRun, but since I am running on lxplus I limited the number of events processed to 1k

MemoryReport> Peak virtual size 20136.1 Mbytes
 Key events increasing vsize:
[2] run: 317696 lumi: 50 event: 37391470  vsize = 14289.8 deltaVsize = 224.012 rss = 7374.43 delta = 143.988
[7] run: 317696 lumi: 50 event: 37681907  vsize = 14833.8 deltaVsize = 320 rss = 7580.88 delta = 206.453
[26] run: 317696 lumi: 50 event: 37558515  vsize = 16045.8 deltaVsize = 640 rss = 7981.98 delta = 401.102
[902] run: 317696 lumi: 50 event: 37422314  vsize = 20136 deltaVsize = 320 rss = 8854.27 delta = 872.289
[58] run: 317696 lumi: 50 event: 37311259  vsize = 16973.8 deltaVsize = 272 rss = 8200.36 delta = 218.383
[983] run: 317696 lumi: 50 event: 37757579  vsize = 20136.1 deltaVsize = 0 rss = 8514.29 delta = -114.879
[982] run: 317696 lumi: 50 event: 37873910  vsize = 20136.1 deltaVsize = 0 rss = 8603.96 delta = -25.2109
[981] run: 317696 lumi: 50 event: 38058508  vsize = 20136.1 deltaVsize = 0.0625 rss = 8629.17 delta = -225.102

There are many MenoryChecks though the job, but most of them are coming from non DQM moculdes though

**Comment by germanfgv** [comment]
The job is supposed to run using 12.3GB maximum, so the 20GB peak virtual size figure seems concerning.

**Comment by jfernan2** [comment]
I understand @germanfgv , however this PR is not "adding" any new process which may increase the memory. Moreover, it is not touching RECO process at all, which is indeed increasing memory.

Since this is the first replay with 12_1_0, (am I right?) may other RECO changes be causing any influence here?

 @srappocc  FYI

**Comment by tvami** [comment]
> I understand @germanfgv , however this PR is not "adding" any new process which may increase the memory. Moreover, it is not touching RECO process at all, which is indeed increasing memory.
> 
> Since this is the first replay with 12_1_0, (am I right?) may other RECO changes be causing any influence here?

@cms-sw/reconstruction-l2  could maybe comment

**Comment by rappoccio** [comment]
@germanfgv @jfernan2 @tvami this is indeed confusing. We expect a relatively large *decrease* in memory consumption naively. Let's wait for input from Reco L2s.

**Comment by slava77** [comment]
> One MET Reco job is using too much memory:
> 
> ```
> PerformanceKill (Exit Code: 50660)
> Error in CMSSW step cmsRun1
> Number of Cores: 8
> Job has exceeded maxPSS: 12300 MB
> Job has PSS: 13540 MB
> ```
> 
> I put logs and tarball in `/afs/cern.ch/user/c/cmst0/public/PausedJobs/DQMseq/job_9738/tarball`

I thought that the limits were still 2.1 GB/core. Why is this one failing at 1.5 ?

**Comment by slava77** [comment]
can somebody summarize what was done on the DQM side?
Do we perhaps see a situation where a bunch of `::one` or legacy modules were removed in DQM and now more modules can run at the same time?

**Comment by tvami** [comment]
I think this is the summary:
 * Created commonReduced which excludes TRK
 * Removed CT-PPS, L1TMon and Tau from common

**Comment by rappoccio** [comment]
I can't think of any changes that would cause this. The only file is here: 

https://github.com/cms-sw/cmssw/pull/35605/files

It's running on fewer modules. However maybe the framework is doing something "clever" and doing them in parallel? Any ideas how we could discern that?

File: etc/ReplayOfflineConfiguration.py [hunk 2]
```diff
@@ -632,7 +638,7 @@
                do_reco=True,
                write_dqm=True,
                alca_producers=["HcalCalNoise"],
-               dqm_sequences=["@common", "@jetmet", "@hcal"],
+               dqm_sequences=["@common", "@jetmet", "@L1TMon", "@hcal"],
```
Comment (germanfgv) [inline etc/ReplayOfflineConfiguration.py:641]: @jfernan2 It looks like the replays is indeed ading DQM sequences to the MET dataset. Is this a mistake?

**Comment by slava77** [comment]
> The job is supposed to run using 12.3GB maximum, so the 20GB peak virtual size figure seems concerning.

I thought that VSIZE is not supposed to matter at all (although there is a history of cutting on it, partly to shoot oneself in the foot).

**Comment by slava77** [comment]
Was there some resolution already to the memory problem?
I did not have time to look at this during the week.

**Comment by germanfgv** [comment]
> Was there some resolution already to the memory problem? I did not have time to look at this during the week.

Hi @slava77. As far as I'm aware, there's no fix yet. Maybe @tvami @rappoccio know something?

**Comment by rappoccio** [comment]
Hi, @germanfgv can we try the replay with the older DQM matrix configuration? Do we need to do something on our side to set this up?

File: etc/ReplayOfflineConfiguration.py [hunk 3]
```diff
@@ -1021,7 +1053,7 @@
                do_reco=True,
                write_reco=True,
                write_dqm=True,
-               dqm_sequences=["@commonSiStripZeroBias", "@ecal", "@hcal", "@muon", "@jetmet"],
+               dqm_sequences=["@commonSiStripZeroBias", "@ecal", "@hcal", "@muon", "@jetmet", "@L1TMon", "@ctpps"],
```
Comment (tvami) [inline etc/ReplayOfflineConfiguration.py:1056]: @jfernan2 here the @L1TMon is after @hcal , is that intentional?

Comment (jfernan2) [reply etc/ReplayOfflineConfiguration.py:1056]: Uhmm, good catch, it is not intentional.... I guess the replay did not catch it if that dataset was not run.... Thanks!

**Comment by jfernan2** [comment]
> @germanfgv I got the usual output from cmsRun, but since I am running on lxplus I limited the number of events processed to 1k
> 
> MemoryReport> Peak virtual size 20136.1 Mbytes Key events increasing vsize: [2] run: 317696 lumi: 50 event: 37391470 vsize = 14289.8 deltaVsize = 224.012 rss = 7374.43 delta = 143.988 [7] run: 317696 lumi: 50 event: 37681907 vsize = 14833.8 deltaVsize = 320 rss = 7580.88 delta = 206.453 [26] run: 317696 lumi: 50 event: 37558515 vsize = 16045.8 deltaVsize = 640 rss = 7981.98 delta = 401.102 [902] run: 317696 lumi: 50 event: 37422314 vsize = 20136 deltaVsize = 320 rss = 8854.27 delta = 872.289 [58] run: 317696 lumi: 50 event: 37311259 vsize = 16973.8 deltaVsize = 272 rss = 8200.36 delta = 218.383 [983] run: 317696 lumi: 50 event: 37757579 vsize = 20136.1 deltaVsize = 0 rss = 8514.29 delta = -114.879 [982] run: 317696 lumi: 50 event: 37873910 vsize = 20136.1 deltaVsize = 0 rss = 8603.96 delta = -25.2109 [981] run: 317696 lumi: 50 event: 38058508 vsize = 20136.1 deltaVsize = 0.0625 rss = 8629.17 delta = -225.102
> 
> There are many MemoryChecks though the job, but most of them are coming from non DQM moculdes though

I have tried the same test removing every DQM sequence:

Without DQM modules:

MemoryReport> Peak virtual size 15632.6 Mbytes
 Key events increasing vsize:
[9] run: 317696 lumi: 50 event: 37536224  vsize = 9958.29 deltaVsize = 256 rss = 4592.9 delta = -249.996
[3] run: 317696 lumi: 50 event: 37753377  vsize = 9346.29 deltaVsize = 256.008 rss = 4842.9 delta = 35.2891
[12] run: 317696 lumi: 50 event: 37753801  vsize = 10662.3 deltaVsize = 704 rss = 4779.89 delta = -63.0039
[145] run: 317696 lumi: 50 event: 37511900  vsize = 13530.3 deltaVsize = 526 rss = 6144.17 delta = 1364.27
[13] run: 317696 lumi: 50 event: 37277143  vsize = 11158.3 deltaVsize = 496 rss = 4993.93 delta = 214.039
[987] run: 317696 lumi: 50 event: 37721567  vsize = 15632.6 deltaVsize = 0 rss = 6768.79 delta = -51.1875
[986] run: 317696 lumi: 50 event: 38107339  vsize = 15632.6 deltaVsize = 0 rss = 6781.12 delta = -38.8516
[985] run: 317696 lumi: 50 event: 38523644  vsize = 15632.6 deltaVsize = 6 rss = 6819.98 delta = 51.5469

Of course, since my lxplus sessions are different and they are shared machines, performance may be different from Tier0 ones.


[1] Config with DQM: /afs/cern.ch/user/j/jfernan/public/testWithDQM.py

[2] Config without DQM: /afs/cern.ch/user/j/jfernan/public/testWithOutDQM.py

**Comment by jfernan2** [comment]
I have fixed the Replay file in this PR with the issues found after and during last Replay:

https://github.com/dmwm/T0/pull/4619#discussion_r747655148
https://github.com/dmwm/T0/pull/4619#discussion_r754530380

**Comment by jfernan2** [comment]
@germanfgv now that memory issues have been clarified and I have fixed differences in the replay config spotted during the review, I do not know if you want to launch another replay or what is the next step. Thanks

**Comment by germanfgv** [comment]
@jfernan2 I'm not sure what you are referring to when you say that the memory issues have been clarified. Did I miss something? could you point me to the discussion?

In any case, if there is a fix to the issue we can test to see if it works. Is it available in one of the pre-releases?

**Comment by jfernan2** [comment]
I am sorry @germanfgv , I thought you were following the Tier0 thread that you initiated:
https://hypernews.cern.ch/HyperNews/CMS/get/tier0-Ops/2314/3/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1.html

**Comment by germanfgv** [comment]
@jfernan2 I think I missed those last few messages. For what I see, there is no changes that need to be tested, just the increase in T0 memory limit (We did discuss this in our T0 internal meeting). In any case, I'd like to confirm that 2GB/core is enough to run the jobs. For that, I'll redeploy the replay.

**Comment by germanfgv** [comment]
run replay please

**Comment by jfernan2** [comment]
Thanks @germanfgv

**Comment by germanfgv** [comment]
After this later retry, we still have one MET promt Reco jobs exceeding the 2GB/core memory limit. @jfernan2 for what I understand, this may improve in the next 12_1_X release, is that right?

**Comment by jfernan2** [comment]
@germanfgv From the RECO side I do not know, from the DQM side there are not improvements expected in memory: this PR is already reducing the number of modules run, hence the memory, w.r.t. 12_0_X

**Comment by slava77** [comment]
> After this later retry, we still have one MET promt Reco jobs exceeding the 2GB/core memory limit. @jfernan2 for what I understand, this may improve in the next 12_1_X release, is that right?

is the log for the job available?
(I've forgotten how to find the replay details on the web with wmstats)

**Comment by jfernan2** [comment]
@slava77  would a backport of this PR reduce the memory?
https://github.com/cms-sw/cmssw/pull/36246
Thanks

**Comment by slava77** [comment]
> @slava77 would a backport of this PR reduce the memory?
> [cms-sw/cmssw#36246](https://github.com/cms-sw/cmssw/pull/36246)

I think so

**Comment by germanfgv** [comment]
> > After this later retry, we still have one MET promt Reco jobs exceeding the 2GB/core memory limit. @jfernan2 for what I understand, this may improve in the next 12_1_X release, is that right?
> 
> is the log for the job available? (I've forgotten how to find the replay details on the web with wmstats)

@slava77  You can find the jobs tarball here:
[C1_vocms047.cern.ch-23926-0-log.tar.gz](https://eoscmsweb.cern.ch/eos/cms/store/logs/prod/recent/PromptReco/PromptReco_Run317696_MET_Tier0_REPLAY_2021_ID211126130552_v1_211126_1319/Reco/C1_vocms047.cern.ch-23926-0-log.tar.gz)

I'll put it in AFS in a few minutes.

**Comment by germanfgv** [comment]
@slava77 In case you prefer:
`/afs/cern.ch/user/c/cmst0/public/PausedJobs/DQMseq/job_23926/tarbal`

**Comment by slava77** [comment]
> @slava77 In case you prefer: `/afs/cern.ch/user/c/cmst0/public/PausedJobs/DQMseq/job_23926/tarball`

Thanks.
I was mainly looking for 
```
Job has exceeded maxPSS: 16000 MB
Job has PSS: 16823 MB
```

**Comment by jfernan2** [comment]
Hi @germanfgv 
I have switched to CMSSW_12_1_1 now that the memory issues have been reduced there

**Comment by germanfgv** [comment]
@jfernan2 great! I'll manually run the test to see if it's enough.

**Comment by germanfgv** [comment]
@jfernan2 The replay finished without problems. The 16GB of memory were enough to reconstruct the MET dataset.

**Comment by Jetmet** [comment]
这是来自QQ邮箱的假期自动回复邮件。您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by jfernan2** [comment]
Hi @germanfgv 
What are the next steps?
Thanks

**Comment by germanfgv** [comment]
Hi @jfernan2, i'll make a clean PR adding these changes to the Replay config and the Production config.

**Comment by tvami** [comment]
Hi @jfernan2 my understanding is that this was added to the prod config, so I believe this PR can be closed, do you agree?

**Comment by Jetmet** [comment]
这是来自QQ邮箱的假期自动回复邮件。您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Pull request closed without merging**